otter/controller.py

```diff
@@ -125,6 +128,66 @@ def obey_config_change(log, transaction_id, config, scaling_group, state,
     return d
 
 
+def force_delete_group(log, trans_id, group):
+    """
+    Delete group even if it has resources by also deleting resources
+
+    :param log: Bound logger
+    :param str trans_id: Transaction ID of request doing this
+    :param otter.models.interface.IScalingGroup scaling_group: the scaling
+        group object
+
+    :return: Deferred that fires with None
+    """
+    if tenant_is_enabled(group.tenant_id, config_value):
+        # For convergence tenants update group status and trigger convergence
+        # TODO: What about status consistency? Will this work if another
+        # process is trying to update status to ERROR at same time?
```

Comment:
@cyli made a suggestion: Instead of relying on single status field, we can have a separate column "deleting" in CASS that will be set True here. The model will still be `update_status` and `get_status` but it will precede DELETING over other states.